odoo/addons/base/views/ir_qweb_widget_templates.xml

```diff
@@ -33,6 +33,9 @@
                 <span>
```

Comment:
And here, below the `<div t-if="city and 'city' in fields" t-attf-class="d-flex align-items-baseline gap-1">` block, we could add another div for the country in case the city is empty and we have the country. So
```xml
<div t-elif="not city and country_id and 'country_id' in fields" t-attf-class="d-flex align-items-baseline gap-1">
    <i t-if="not options.get('no_marker')" class="fa fa-map-marker fa-fw" role="img" aria-label="Address" title="Address"/>
    <span>
         <div>
            <t t-if="object.state_id and object.country_id.state_required">
                <span itemprop="addressState" t-esc="object.state_id.code"/>,
            </t>
            <span itemprop="addressCountry" t-esc="country_id"/>
        </div>
    </span>
</div>
```